src/modules/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -1,13 +1,60 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 
-import { StacheNav, StacheNavLink } from '../nav';
+import { StacheNav, StacheNavLink, StacheNavService } from '../nav';
 
 @Component({
   selector: 'stache-breadcrumbs',
   templateUrl: './breadcrumbs.component.html',
   styleUrls: ['./breadcrumbs.component.scss']
 })
-export class StacheBreadcrumbsComponent implements StacheNav {
+export class StacheBreadcrumbsComponent implements StacheNav, OnInit {
   @Input()
   public routes: StacheNavLink[];
+
+  public constructor(
+    private navService: StacheNavService) { }
+
+  public ngOnInit(): void {
+    if (!this.routes) {
+      const activeRoutes = this.navService.getActiveRoutes();
+      this.routes = this.filterRoutes(activeRoutes);
+    }
+  }
+
+  private filterRoutes(activeRoutes: StacheNavLink[]): StacheNavLink[] {
+    const activeUrl = this.navService.getActiveUrl();
+    const root = activeRoutes[0];
+    let routes: StacheNavLink[] = [];
+
+    routes.push({
+      name: 'Home',
+      path: '/'
+    });
+
+    if (root.path !== '/') {
+      routes.push({
+        name: root.name,
+        path: root.path
+      });
+    }
+
+    const findActiveRoute = (r: any[]) => {
```

Comment:
I know why you called this `r`, but can we name it something more descriptive? `interiorRoutes` or `routesArray` or something?